kudosaurus.go

```diff
@@ -0,0 +1,52 @@
+package kudosaurus
+
+// Kudo - kudo entity
+type Kudo struct {
+	ID          string
+	TeamID      string
+	Text        string
+	PersonFrom  Person
+	PersonTo    Person
+	IsPublished bool
+}
+
+// Person - person entity
+type Person struct {
+	ID         string
+	Name       string
+	Kudos      []Kudo
+	KudosGiven []Kudo
+	KudosLeft  int
+	IsAdmin    bool
+}
+
+// Team - team entity
+type Team struct {
+	ID       string
+	Name     string
+	People   []Person
+	Settings Settings
+}
+
+// Settings - runtime settings for a team
+type Settings struct {
+	TeamID          string
+	TargetChannelID string
+}
+
+// Store - stores data
+type Store interface {
+	PutPerson(id string, name string, teamID string, teamName string) (Person, error)
```

Comment:
I do not see `context.Context` being passed anywhere in here. Passing ctx all the way down from HTTP handler to database implementation allows early cancelation of long running tasks and is good practice in general. You could have some important information injected into the context.

Reply:
I'll have to look into the entire concept of contexts - I don't think it has a good equivalent in anything I used so far. Or I might not see the similarity.

Reply:
I do not think there is actually. And also, many different usages of context are very opinionated. Although, using context as a circuit breaker is generally a way to go and it's very handy.